include/zephyr/sys/dlist.h

```diff
@@ -492,6 +492,15 @@ static inline void sys_dlist_insert_at(sys_dlist_t *list, sys_dnode_t *node,
 	}
 }
 
+static inline void sys_dlist_dequeue(sys_dnode_t *node)
```

Comment:
Isn't this equivalent to `sys_dlist_remove()`?

Why not simply remove the `init()` from here? It is

https://github.com/zephyrproject-rtos/zephyr/blob/049b243b51f354c27273094a798ad2c9f34481c1/include/zephyr/sys/dlist.h#L511

Reply:
The `init()` in the `remove()` case allows us to determine whether the node is part of any doubly linked list--see _sys_dnode_is_linked()_. However, there are cases where a node can be removed from one list only to be immediately inserted into a doubly linked list elsewhere. When such cases on a critical path, it may be desirable to have an option that does not re-initialize the node.

Reply:
maybe dequeue is the wrong name here, you are calling a generic api based on where you are using it (in priority queues), would be `sys_dlist_remove_no_init()` be more appropriate? :-) .

Reply:
This cleaned up branch version somehow lost that warning. I'll definitely add it back.

Reply:
Naming things is hard. As long as the documentation is clear